src/main/java/edu/wisc/my/messages/model/Message.java

```diff
@@ -318,50 +318,44 @@ public void setConfirmButton(ActionButton confirmButton) {
   }
 
   @JsonIgnore
-  public boolean isValidToday(){
-
-   boolean isValidToday = false;
-   Date today = new Date();
-   
-   // Placeholder date format string, 
-   // can be overridden with an application property.
-   String formatString = "yyyy-mm-dd";
-
-   try{ 
-   //If there's no date specified, the message is valid.
-   if(StringUtils.isBlank(getGoLiveDate()) && 
-     StringUtils.isBlank(getExpireDate())) {
-       return true;
-     }
-
-  SimpleDateFormat defaultFormatter = new SimpleDateFormat(formatString);
-
-  String goLive = getGoLiveDate();
-  String expiration = getExpireDate();
-  
-  // If the message has no goLiveDate, we presume it to have been valid 
-  // since the beginning of time. Or 1977. Whichever came first.
-  String startDateString= (StringUtils.isNotBlank(goLive)) ? 
-       goLive : "1977-08-16";
-
-  // If the message has no expirationDate, we presume it will be valid until 
-  // the end of time. Or until we have a Y4K problem. Whichever comes first.       
-  String endDateString= (StringUtils.isNotBlank(expiration)) ?
-       expiration : "3999-12-31";
-
-   Date startDate = defaultFormatter.parse(startDateString);
-   Date endDate = defaultFormatter.parse(endDateString);
-
-    if(today.after(startDate) && today.before(endDate)) {
-      isValidToday = true;
-    }
-   } catch (Exception e) {
-     logger.warn("DATE ERROR " + this.id + " " + e.getMessage());
-     isValidToday =  false;
-   }
+  public boolean isValidToday() {
+    LocalDateTime now = LocalDateTime.now();
+
+    try {
+      // if the message is premature, it is not valid.
+      // go live date might be time specified
+      if (StringUtils.isNotBlank(goLiveDate) && goLiveDate.contains("T")
```

Comment:
could `StringUtils.isNotBlank(goLiveDate) && goLiveDate.contains("T")` be checked a single time before any of the `LocalDate.parse(goLiveDate)`? Is it needed at all? It looks like invalid date exception is caught and gracefully processed.